tls/tls_ka.go

```diff
@@ -102,9 +102,9 @@ func (ka *dheKeyAgreement) DHParams() *keys.DHParams {
 	}
 	if ka.yServer != nil {
 		out.ServerPublic = new(big.Int).Set(ka.yServer)
-	}
-	if ka.yServer.Cmp(ka.yOurs) == 0 && ka.xOurs != nil {
-		out.ServerPrivate = new(big.Int).Set(ka.xOurs)
+		if ka.yOurs != nil && ka.yServer.Cmp(ka.yOurs) == 0 && ka.xOurs != nil {
```

Comment:
Shouldn't we check both ` ka.yOurs != nil` and `ka.xOurs != nil` before checking `ka.yServer.Cmp(ka.yOurs)`?